.github/workflows/ci.yaml

```diff
@@ -0,0 +1,18 @@
+---
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  statics:
+    name: Static checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: pre-commit/action@v1.0.1
```

Comment:
I would see this as a temporary step to avoid to vast scope for single PR. 

From my experience, I would suggest migrating to Github Actions. There are multiple reasons:
- stability
- better way of defining workflows (for example using matrices) 
- better integration with Github

We experience too much pain with Travis in Apache Airflow. And since Github Actions are getting a lot of love from Github I would say that this is a better way forward 😉

Reply:
I see. Thanks for clarifing.

From my point of view, Travis has stability, matrices, and sufficiently good integration with GitHub. I had no problems with it so far. It would be good if you can share an article or a talk what kind of problem you had with it — maybe it would prevent Kopf from stepping into the same problems.

For now, sorry, I cannot take this as an objective argument for the switch. I take this as your opinion based on your experience and your preferences. Sadly, I cannot form my own opinion right now — it would require time (see another comment — to play with GitHub Actions myself in a separate repo).

For this PR, I suggest to keep Travis CI as the one and the only CI solution in the project. When and if the migration to GitHub Action happens after a thorough consideration, this should be done consistently for all build steps (static analysis, unit-tests, python versions, integration tests, k8s versions (k3s/k3d), lengthy nightly tests (minikube), etc). Consistency has value.

I've created #554 to consider GitHub Actions (and I see one big problem there already — free minutes limit).

---

Regarding `pre-commit` itself: I like it, I am all for it, but with common sense applied: the little issues in styling/formatting should not prevent the contributors to contribute. 

It is fine if the codebase is not perfectly styled (again, within common sense). The project is not that popular to have thousands of developers fighting over the code-style, so that it could sacrifice some percentage of them. 

I abandoned my attempts to contribute to some good projects because the contribution experience turned into fighting with spaces and commas of their style checkers. Or worse, with code uglifiers like Black (I will be strongly against that kind of things).

I don't want people to feel the same here. Every contribution matters! (but again: common sense).

Reply:
> I abandoned my attempts to contribute to some good projects because the contribution experience turned into fighting with spaces and commas of their style checkers. Or worse, with code uglifiers like Black (I will be strongly against that kind of things).

From the maintainer point of view - code quality is as important as number of contributors. The more is done automatically (formatting, linting, checking etc) the more attention maintainer can pay to review the changes. Avoiding using mainstream and well-known approaches (black, mypy, pylint) results in code that is not consistent, thus hard to read. But I fully understand you, it's your project.

> I don't want people to feel the same here. Every contribution matters! (but again: common sense).

I think what matters is how much help people get from maintainers. In Airflow we had 40+ checks including, linting and checking some wording and missing tests. And still, we see a lot of people who answer to good first issues and are surprised by the ease of contributing.

Reply:
That’s exactly my point: a project with 7k forks and 50 PRs/week has slightly different rules of play, requirements, and solutions. 

When and if Kopf will be at these levels, I will be actively advertising for Black myself — because it makes sense and has reasons: the disadvantages of not using it outweigh the disadvantages of using it. 

Until then, let’s keep it simple.